client-side/src/graphs/graphs.jsx

```diff
@@ -1,125 +1,92 @@
-
-    let webSite = websites.data.websites.map((obj, index) => ({ ...obj, color: colors[index] }))
-    const u = users.data.users.find(u => u.email === user.email)
-    console.log({ u });
-    
-    const totalActivityTimeByWebsite = {};
+    if (websites.error || users.error) return <p>Error</p>;
+
+    function getWebsites(dateStart, dateEnd) {
+        let website = websites?.data?.websites?.map((obj, index) => ({ ...obj, color: colors[index] }))
+        const u = users?.data?.users?.find(u => u.email === user.email)
+
+        const totalActivityTimeByWebsite = {};
+
+        u.visitsWebsites.forEach(visitedWebsite => {
+            const websiteName = visitedWebsite.websiteId.name;
+            const activityTime = visitedWebsite.visitsTime.reduce((total, visit) => {
+                let date = new Date(visit.visitDate)
+                date = formatDate(date)
+                if (date >= dateStart && date <= dateEnd) {
+                    return total + Number(visit.activityTime);
+                }
+                return total;
+            }, 0);
+
+            if (totalActivityTimeByWebsite[websiteName]) {
+                totalActivityTimeByWebsite[websiteName] += activityTime;
+            } else {
+                totalActivityTimeByWebsite[websiteName] = activityTime;
+            }
+        });
 
-    u.visitsWebsites.forEach(visitedWebsite => {
-        const websiteName = visitedWebsite.websiteId.name;
-        const activityTime = visitedWebsite.visitsTime.reduce((total, visit) => {
-            let date = new Date(visit.visitDate)
-            year = date.getFullYear();
-            month = ('0' + (date.getMonth() + 1)).slice(-2);
-            day = ('0' + date.getDate()).slice(-2);
-            date = `${year}-${month}-${day}`
-            if (date >= dateStart && date <= dateEnd) {
-                return total + Number(visit.activityTime);
+        website.forEach(site => {
+            const websiteName = site.name;
+            if (totalActivityTimeByWebsite[websiteName]) {
+                site.time = totalActivityTimeByWebsite[websiteName];
             }
-            return total;
-        }, 0);
+        });
+
+        return website
+    }
+    let website = getWebsites(dateStart, dateEnd)
 
-        if (totalActivityTimeByWebsite[websiteName]) {
-            totalActivityTimeByWebsite[websiteName] += activityTime;
-        } else {
-            totalActivityTimeByWebsite[websiteName] = activityTime;
-        }
-    });
 
-    webSite.forEach(site => {
-        const websiteName = site.name;
-        if (totalActivityTimeByWebsite[websiteName]) {
-            site.time = totalActivityTimeByWebsite[websiteName];
-        }
-    });
+    const otherProps = {
+        width: 400,
+        height: 200,
+        sx: {
+            [`.${ChartsLegend.root}`]: {
+                transform: 'translate(20px, 0)',
+            },
+        },
+    };
+   
+    VisitedWebsitesComponent.propTypes = {
```

Comment:
prop types should be defined outside the component, in the end of the file, before the return statment